fpdf/table.py

```diff
@@ -268,23 +280,124 @@ def _render_table_cell(
                 if style
                 else FontFace(fill_color=self._cell_fill_color)
             )
-        with self._fpdf.use_font_face(style):
-            page_break, cell_height = self._fpdf.multi_cell(
-                w=col_width,
-                h=row_height,
-                txt=cell.text,
-                max_line_height=self._line_height,
-                border=self.get_cell_border(i, j),
-                align=text_align,
-                new_x="RIGHT",
-                new_y="TOP",
-                fill=fill,
-                markdown=self._markdown,
-                output=MethodReturnValue.PAGE_BREAK | MethodReturnValue.HEIGHT,
-                wrapmode=self._wrapmode,
-                **kwargs,
+
+        if cell.padding:
+            from fpdf.fpdf import get_padding_tuple
+
+            padding = get_padding_tuple(cell.padding)
+        else:
+            padding = self._padding
+
+        if cell.v_align:
+            v_align = cell.v_align
+        else:
+            v_align = self._v_align
+
+        # place cursor (required for images after images)
+        cell_widhts = [self._get_col_width(i, jj) for jj in range(j)]
+        cell_x = sum(cell_widhts)
+        self._fpdf.set_x(self._fpdf.l_margin + cell_x + self._gutter_width * j)
+
+        # render cell border and background
+
+        # if cell_height is defined, that means that we already know the size at which the cell will be rendered
+        # so we can draw the borders now
+        #
+        # If cell_height is None then we're still in the phase of calculating the height of the cell meaning that
+        # we do not need to render and borders yet.
+        if cell_height is not None:
+            with self._fpdf.use_font_face(style):
+                x1 = self._fpdf.x
+                y1 = self._fpdf.y
+                x2 = x1 + col_width
+                y2 = y1 + cell_height
+
+                self._fpdf._draw_box(
+                    x1, y1, x2, y2, border=self.get_cell_border(i, j), fill=fill
+                )
+
+        # render image
+
+        if cell.img:
+            x, y = self._fpdf.x, self._fpdf.y
+
+            # if cell_height is None or width is given then call image with h=0
+            # calling with h=0 means that the image will be rendered with an auto determined height
+            auto_height = cell.img_fill_width or cell_height is None
+
+            # apply padding
+            self._fpdf.x += padding[3]
+            self._fpdf.y += padding[0]
+
+            image = self._fpdf.image(
+                cell.img,
+                w=col_width - padding[1] - padding[3],
+                h=0 if auto_height else cell_height - padding[0] - padding[2],
+                keep_aspect_ratio=True,
             )
-        return page_break, max(img_height, cell_height)
+
+            img_height = image.rendered_height
+
+            if img_height + y > self._fpdf.page_break_trigger:
+                page_break_image = True
```

Comment:
Interesting. This is a different approach than the one currently in the code.

Overall, we rely on the output of `multi_cell()` with `output=MethodReturnValue.PAGE_BREAK | MethodReturnValue.HEIGHT` to figure the actual row heigh & potential page break.

At the moment, the actual code does not detect that an image triggers a page break on the "first pass"
(when `_render_table_cell()` is called from `_get_row_layout_info()`),
but because the image height is included in `row_layout_info.height`, it still works fine
and a page jump is performed on the "second pass"
(when `_render_table_cell()` is called from `_render_table_row()`)

Your approach of detecting a page jump seems totally fine / valid,
but we need to make sure it produces the same output as currently,
and that it correctly handles cells with bot image & text (_cf._ `test_table_with_images_and_text()`)